connector/otlpjsonconnector/logs.go

```diff
@@ -50,16 +50,26 @@ func (c *connectorLogs) ConsumeLogs(ctx context.Context, pl plog.Logs) error {
 			for k := 0; k < logRecord.LogRecords().Len(); k++ {
 				lRecord := logRecord.LogRecords().At(k)
 				token := lRecord.Body()
-				var l plog.Logs
-				l, err := logsUnmarshaler.UnmarshalLogs([]byte(token.AsString()))
-				if err != nil {
-					c.logger.Error("could not extract logs from otlp json", zap.Error(err))
+
+				// Check if the "resourceLogs" key exists in the JSON data
+				if logRegex.MatchString(token.AsString()) {
+					var l plog.Logs
+					l, err := logsUnmarshaler.UnmarshalLogs([]byte(token.AsString()))
+					if err != nil {
+						c.logger.Error("could not extract logs from otlp json", zap.Error(err))
+						continue
+					}
+					err = c.logsConsumer.ConsumeLogs(ctx, l)
+					if err != nil {
+						c.logger.Error("could not consume logs from otlp json", zap.Error(err))
+					}
+				} else if metricRegex.MatchString(token.AsString()) || traceRegex.MatchString(token.AsString()) {
 					continue
+				} else {
+					c.logger.Error("Invalid otlp payload")
```

Comment:
It's not strictly against a principle of connectors, but I would highlight some considerations.

The user can opt to place this connector in any or all of its supported signal combinations. What you're proposing works if and only if it is used in `logs -> logs`, so such an option would be meaningless if a user only uses it in `logs -> metrics` or `logs -> traces`.

That said, it could just be an option that is documented to only apply if used in `logs -> logs`. This still feels a bit awkward to me because of the asymmetry between the signals. A similar option would be to introduce a setting like `err_pipeline` (or `err_pipelines`) which must be log pipelines. The user could choose to provide a dedicated route for non-conforming inputs, or they can explicitly flow them into the same logs pipeline to which valid otlpjson log inputs would flow.

Reply:
Thank's! I agree this could complicate things without adding much value.